server.go

```diff
@@ -1683,6 +1702,138 @@ func (s *Server) executeDropUserStatement(q *influxql.DropUserStatement, user *U
 	return &Result{Err: s.DeleteUser(q.Name)}
 }
 
+func (s *Server) executeShowSeriesStatement(stmt *influxql.ShowSeriesStatement, database string, user *User) *Result {
+	// Find the database.
+	db := s.database(database)
```

Comment:
this entire method should actually get an `RLock` on the server so the method for `.databases()` isn't necessary.